scripts/idl/test_xml_parser.py

```diff
@@ -0,0 +1,197 @@
+
+    def testCluster(self):
+        idl = XmlToIdl('''<?xml version="1.0"?>
+            <!-- Comments should be supported -->
+            <configurator>
+              <cluster>
+                <name>Test Cluster</name>
+                <code>0x1234</code>
+
+                <attribute side="server" code="11" type="INT32U" min="0" max="2" isNullable="true" reportable="true" writable="false">SomeIntAttribute</attribute>
+
+                <attribute side="server" code="22" define="SOME_DEFINE" type="INT8U" min="0" max="10" reportable="true" default="0" writable="true" optional="true">
+                    <description>AttributeWithAccess</description>
+                    <access op="read" role="operate" />
+                    <access op="write" role="manage" />
+                </attribute>
+
+                <command source="client" code="33" name="GetSomeData" response="GetSomeDataResponse" optional="true">
+                    <description>This is just a test: client to server</description>
+                    <access op="invoke" role="administer" />
+                    <arg name="firstInput" type="INT8U" />
+                    <arg name="secondInput" type="INT16U" />
+                </command>
+
+                <command source="server" code="0x44" name="GetSomeDataResponse" disableDefaultResponse="true" optional="true">
+                    <description>Reply from server</description>
+                    <arg name="dataPoint1" type="INT8U" />
+                    <arg name="dataPoint2" type="INT8U" optional="true" />
+                </command>
+              </cluster>
+            </configurator>
+        ''')
+        self.assertEqual(idl,
+                         Idl(clusters=[
+                             Cluster(
+                                 side=ClusterSide.CLIENT,
+                                 name='TestCluster',
+                                 code=0x1234,
+                                 attributes=[
+                                     Attribute(definition=Field(data_type=DataType(name='INT32U'), code=11, name='SomeIntAttribute',
+                                                                attributes={FieldAttribute.NULLABLE}), tags={AttributeTag.READABLE},
+                                               readacl=AccessPrivilege.VIEW, writeacl=AccessPrivilege.OPERATE),
+                                     Attribute(definition=Field(data_type=DataType(name='INT8U'), code=22, name='AttributeWithAccess',
+                                                                attributes={FieldAttribute.OPTIONAL}),
+                                               tags={AttributeTag.READABLE, AttributeTag.WRITABLE}, readacl=AccessPrivilege.OPERATE,
+                                               writeacl=AccessPrivilege.MANAGE)
+                                 ],
+                                 structs=[
+                                     Struct(name='GetSomeDataRequest',
```

Comment:
A command definition may look like a struct, but it isn't a struct. E.g a struct has no 'ID', but a command does.

Reply:
This is the current IDL format based on how zapt is being parsed.
Commands contain the id for the request (and request struct is plain), response structures contain the response id.

This is not very clean and intuitive and I believe we should update it somehow - maybe commands should contain both request and response codes and structs should become simple in/out data. I am not sure how to structure that but know it needs to be done.

I believe this should be as a followup as this PR is intended to only match current structure of IDL. Changing IDL can be done separately.

Reply:
Follow-up is fine.